api/v1/announcements/PKPAnnouncementHandler.php

```diff
@@ -341,4 +306,87 @@ public function delete($slimRequest, $response, $args)
 
         return $response->withJson($announcementProps, 200);
     }
+
+    /**
+     * Get the context id or site-wide context id of the current request
+     */
+    protected function getContextId(): int
```

Comment:
This may need to change with https://github.com/pkp/pkp-lib/pull/9076#discussion_r1228787990 -- but I think that work is needing revision, so just FYI for now.

Reply:
Ah yeah, I had the same thought. It's a tricky problem. Good luck! :)

Reply:
Changed to using `null` as discussed in the comment below.